clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -206,25 +201,27 @@ public boolean assignFromUser(Set<TopicPartition> partitions) {
      *
      * @return true if assignments matches subscription, otherwise false
      */
-    public boolean assignFromSubscribed(Collection<TopicPartition> assignments) {
+    public synchronized boolean assignFromSubscribed(Collection<TopicPartition> assignments) {
         if (!this.partitionsAutoAssigned())
             throw new IllegalArgumentException("Attempt to dynamically assign partitions while manual assignment in use");
 
         Predicate<TopicPartition> predicate = topicPartition -> {
-            if (this.subscribedPattern != null) {
-                boolean match = this.subscribedPattern.matcher(topicPartition.topic()).matches();
-                if (!match) {
-                    log.info("Assigned partition {} for non-subscribed topic regex pattern; subscription pattern is {}",
-                            topicPartition,
-                            this.subscribedPattern);
-                }
-                return match;
-            } else {
-                boolean match = this.subscription.contains(topicPartition.topic());
-                if (!match) {
-                    log.info("Assigned partition {} for non-subscribed topic; subscription is {}", topicPartition, this.subscription);
+            synchronized (SubscriptionState.this) {
```

Comment:
Spotbugs throw an error for this but I don't think it's accurate since we synchronize the whole method and this predicate will be executed there. Should I add this to the excludes?

Reply:
Actually I'm not sure why we need this additional block if we are already holding the lock. The predicate wouldn't be used after this method returns.

Reply:
![image](https://user-images.githubusercontent.com/1820518/56370785-28763880-61fc-11e9-88a7-a3d5a3f8d9e5.png)
Basically this is why I added it, I think spotbugs incorrectly detects this. I hope I can make an exception in spotbugs for this method.